Koalas GeoSeries symmetric_difference bug

**Describe the bug**

Koalas GeoSeries symmetric_difference got exception while postgis return normal result.

**To Reproduce**

Koalas GeoSeries test code:

```python
>>> from arctern_spark.geoseries import GeoSeries
>>> col1=['POINT (0 0)']
>>> col2=['POINT (0 0)']
>>> geo1=GeoSeries(col1)
>>> geo2=GeoSeries(col2)
>>> geo1.symmetric_difference(geo2)
20/07/11 17:29:40 ERROR Executor: Exception in task 5.0 in stage 2.0 (TID 24)12]
java.lang.IllegalArgumentException: Empty Points cannot be represented in WKB
```

postgis test sql:

```sql
postgres=# select st_symdifference('POINT (0 0)'::geometry, 'POINT (0 0)'::geometry);
  st_symdifference  
--------------------
 010700000000000000
(1 row)

postgres=# select st_astext(st_symdifference('POINT (0 0)'::geometry, 'POINT (0 0)'::geometry));
        st_astext         
--------------------------
 GEOMETRYCOLLECTION EMPTY
(1 row)


```

**Any logs, error output, snapshot, etc?**

log was below:

```
20/07/11 17:29:40 ERROR Executor: Exception in task 5.0 in stage 2.0 (TID 24)12]
java.lang.IllegalArgumentException: Empty Points cannot be represented in WKB
        at org.locationtech.jts.io.WKBWriter.writePoint(WKBWriter.java:334)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:308)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:290)
        at org.apache.spark.sql.arctern.GeometryUDT$.ToWkb(GeometryUDT.scala:47)
        at org.apache.spark.sql.arctern.GeometryUDT$.GeomSerialize(GeometryUDT.scala:38)
        at org.apache.spark.sql.arctern.GeometryUDT.GeomSerialize(GeometryUDT.scala)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage5.processNext(Unknown Source)
        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
        at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:132)
        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
        at org.apache.spark.scheduler.Task.run(Task.scala:127)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
20/07/11 17:29:41 WARN TaskSetManager: Lost task 5.0 in stage 2.0 (TID 24, spark-slave, executor driver): java.lang.IllegalArgumentException: Empty Points cannot be represented in WKB
        at org.locationtech.jts.io.WKBWriter.writePoint(WKBWriter.java:334)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:308)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:290)
        at org.apache.spark.sql.arctern.GeometryUDT$.ToWkb(GeometryUDT.scala:47)
        at org.apache.spark.sql.arctern.GeometryUDT$.GeomSerialize(GeometryUDT.scala:38)
        at org.apache.spark.sql.arctern.GeometryUDT.GeomSerialize(GeometryUDT.scala)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage5.processNext(Unknown Source)
        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
        at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:132)
        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
        at org.apache.spark.scheduler.Task.run(Task.scala:127)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

20/07/11 17:29:41 ERROR TaskSetManager: Task 5 in stage 2.0 failed 1 times; aborting job
20/07/11 17:29:41 WARN TaskSetManager: Lost task 38.0 in stage 2.0 (TID 62, spark-slave, executor driver): TaskKilled (Stage cancelled)
20/07/11 17:29:41 WARN TaskSetManager: Lost task 40.0 in stage 2.0 (TID 64, spark-slave, executor driver): TaskKilled (Stage cancelled)
20/07/11 17:29:41 WARN TaskSetManager: Lost task 37.0 in stage 2.0 (TID 61, spark-slave, executor driver): TaskKilled (Stage cancelled)
20/07/11 17:29:41 WARN TaskSetManager: Lost task 39.0 in stage 2.0 (TID 63, spark-slave, executor driver): TaskKilled (Stage cancelled)
20/07/11 17:29:41 WARN TaskSetManager: Lost task 41.0 in stage 2.0 (TID 65, spark-slave, executor driver): TaskKilled (Stage cancelled)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/arctern_spark-0.0.0-py3.7.egg/arctern_spark/geoseries.py", line 282, in __repr__
    pser = wkt_ks._to_internal_pandas()
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/databricks/koalas/series.py", line 5156, in _to_internal_pandas
    return first_series(self._internal.to_pandas_frame)
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/databricks/koalas/utils.py", line 465, in wrapped_lazy_property
    setattr(self, attr_name, fn(self))
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/databricks/koalas/internal.py", line 755, in to_pandas_frame
    pdf = sdf.toPandas()
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/pyspark-3.0.0-py3.7.egg/pyspark/sql/pandas/conversion.py", line 138, in toPandas
    pdf = pd.DataFrame.from_records(self.collect(), columns=self.columns)
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/pyspark-3.0.0-py3.7.egg/pyspark/sql/dataframe.py", line 596, in collect
    sock_info = self._jdf.collectToPython()
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/py4j-0.10.9-py3.7.egg/py4j/java_gateway.py", line 1305, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/pyspark-3.0.0-py3.7.egg/pyspark/sql/utils.py", line 131, in deco
    return f(*a, **kw)
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/py4j-0.10.9-py3.7.egg/py4j/protocol.py", line 328, in get_return_value
    format(target_id, ".", name), value)
py4j.protocol.Py4JJavaError: An error occurred while calling o344.collectToPython.
: org.apache.spark.SparkException: Job aborted due to stage failure: Task 5 in stage 2.0 failed 1 times, most recent failure: Lost task 5.0 in stage 2.0 (TID 24, spark-slave, executor driver): java.lang.IllegalArgumentException: Empty Points cannot be represented in WKB
        at org.locationtech.jts.io.WKBWriter.writePoint(WKBWriter.java:334)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:308)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:290)
        at org.apache.spark.sql.arctern.GeometryUDT$.ToWkb(GeometryUDT.scala:47)
        at org.apache.spark.sql.arctern.GeometryUDT$.GeomSerialize(GeometryUDT.scala:38)
        at org.apache.spark.sql.arctern.GeometryUDT.GeomSerialize(GeometryUDT.scala)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage5.processNext(Unknown Source)
        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
        at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:132)
        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
        at org.apache.spark.scheduler.Task.run(Task.scala:127)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

Driver stacktrace:
        at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2023)
        at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:1972)
        at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:1971)
        at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
        at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
        at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1971)
        at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:950)
        at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:950)
        at scala.Option.foreach(Option.scala:407)
        at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:950)
        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2203)
        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2152)
        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2141)
        at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
        at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:752)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2093)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2114)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2133)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2158)
        at org.apache.spark.rdd.RDD.$anonfun$collect$1(RDD.scala:1004)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
        at org.apache.spark.rdd.RDD.withScope(RDD.scala:388)
        at org.apache.spark.rdd.RDD.collect(RDD.scala:1003)
        at org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:385)
        at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:3448)
        at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3616)
        at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:100)
        at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:160)
        at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:87)
        at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:763)
        at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
        at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3614)
        at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:3445)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: Empty Points cannot be represented in WKB
        at org.locationtech.jts.io.WKBWriter.writePoint(WKBWriter.java:334)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:308)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:290)
        at org.apache.spark.sql.arctern.GeometryUDT$.ToWkb(GeometryUDT.scala:47)
        at org.apache.spark.sql.arctern.GeometryUDT$.GeomSerialize(GeometryUDT.scala:38)
        at org.apache.spark.sql.arctern.GeometryUDT.GeomSerialize(GeometryUDT.scala)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage5.processNext(Unknown Source)
        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
        at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:132)
        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
        at org.apache.spark.scheduler.Task.run(Task.scala:127)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        ... 1 more
```

--- a/scala/src/main/scala/org/apache/spark/sql/arctern/expressions/Functions.scala
+++ b/scala/src/main/scala/org/apache/spark/sql/arctern/expressions/Functions.scala
@@ -733,7 +733,7 @@ case class ST_SymDifference(inputsExpr: Seq[Expression]) extends ST_BinaryOp {
 
   override def rightExpr: Expression = inputsExpr(1)
 
-  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = codeGenJob(ctx, ev, (left, right) => s"$left.symDifference($right)")
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = codeGenJob(ctx, ev, (left, right) => s"$left.symDifference($right).isEmpty() ? new org.locationtech.jts.geom.GeometryFactory().createGeometryCollection() : $left.symDifference($right)")
 
   override def dataType: DataType = new GeometryUDT
 

--- a/scala/src/test/scala/FunctionsTest.scala
+++ b/scala/src/test/scala/FunctionsTest.scala
@@ -2159,6 +2159,7 @@ class FunctionsTest extends AdapterTest {
   test("ST_SymDifference") {
     val data = Seq(
       Row(GeometryUDT.FromWkt("LINESTRING (0 0,5 0)"), GeometryUDT.FromWkt("LINESTRING (4 0,6 0)")),
+      Row(GeometryUDT.FromWkt("POINT (0 0)"), GeometryUDT.FromWkt("POINT (0 0)")),
     )
 
     val schema = StructType(Array(StructField("left_geo", new GeometryUDT, nullable = true), StructField("right_geo", new GeometryUDT, nullable = true)))
@@ -2173,6 +2174,7 @@ class FunctionsTest extends AdapterTest {
     val collect = rst.collect()
 
     assert(collect(0).getAs[GeometryUDT](0).toString == "MULTILINESTRING ((0 0, 4 0), (5 0, 6 0))")
+    assert(collect(1).getAs[GeometryUDT](0).toString == "GEOMETRYCOLLECTION EMPTY")
 
     val rst2 = df.select(st_symdifference(col("left_geo"), col("right_geo")))
     rst2.show(false)
@@ -2180,6 +2182,7 @@ class FunctionsTest extends AdapterTest {
     val collect2 = rst2.collect()
 
     assert(collect2(0).getAs[GeometryUDT](0).toString == "MULTILINESTRING ((0 0, 4 0), (5 0, 6 0))")
+    assert(collect2(1).getAs[GeometryUDT](0).toString == "GEOMETRYCOLLECTION EMPTY")
   }
 
   test("ST_SymDifference-Null") {

